### PR TITLE
feat: teams page

### PR DESCRIPTION
**Pull request by joanna-pagepro**
https://tw.auroracreation.com/app/tasks/15641315

Description
As I user I want to visit a page with a list of all Teams within the Colony, to get more knowledge about them. I can quickly navigate to other pages, like Activity feed, with the Team filter activated.

There's no Teams page in old UI.

Acceptance criteria

General requirements:
  - UI should match the Figma design
  - Teams page is linked from the Members tab
  - dipslay page Title and Breadrcrubs
    - no need for filtering by Teams on this page, just display "All teams"
  - an option to quickly create new team using Create a team button
    - triggers action panel witt preselected "Creat new team" actio
  
Team Cards:

- Team cards are ordered by highest to lowest reputation percentage by default.
- There is no "load more" button, just a continual scroll if there are more teams in the individual colony.
- Team name
  - if the team name is long and doesn’t fit, it becomes truncated.
- Team pill
- Team funds
  - tooltip on hover <- UPDATE 22.12
  - display funds in native token
  - this will need to work with currency switcher, right now we are not supprting currencies
- Team reputation percentage
  - tooltip on hover <- UPDATE 22.12
- Team description
  - it's fetched from Team purpose field that is set during team creation
  - If there is no description provided for the team the box height of the description content remains fixed so the heights of the boxes remain equal.
    - This is applicable for desktop, tablet and mobile.
  - If the content is less than 3 sentences, the box height of the description content remains fixed so the heights of the boxes remain equal.
    - This is applicable for desktop, tablet and mobile.
  - Max description length is max three sentences for each card. If more content is available, the content is truncated with three ellipses shown.
    - This is the same for desktop and tablet only.
  - On mobile the full description is shown and not truncated since the cards are stacked.
    - There is no tool tip to see all the content on mobile.
  - If the content is truncated, the user can hover over the truncated description and the complete content will display.
    - This is applicable for tablet but not on hover, but when the user presses the content block.

- Member's avatars
  - similar component is used on Dashboard page
  - Team cards show a maximum of 5 user avatars.
  - If there are more than 5, a default card showing the rest of the members in a total.
  
- Agreements button & Activity feed button
  - Agreements links the user to the decisions page with the specific team filtered.
    - This page is not ready yet
  - Activity feed links to the user to the main activity page with the specific team filtered.
  - If the team does not have any activity or decisions yet, the text links do not show.

- Menu button:

  - edit team: triggers edit team with prefilled team
  - view team member: goes to filtered members list with appropriate team selected
  - view team activity: goes to the activity feed with the specific team filter applied
  - view team funds: Takes user to the teams funds page with the specific team filtered
  - view team dicisions: Takes user to the decisions page with the specific team filter applied
  - view team dashboard: Tasks user to the main dashboard of the Colony with the specific team filter applied

Out of scope

- decision/agreements page
- any fixes to the page linked from menu
- Search & sorting

Design

New layout:
https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-V5?type=design&node-id=9707-64609&mode=design&t=jzeXwFTUtZKkBmIT-4

image

### CODE REVIEW

**Review by arrenv — CHANGES_REQUESTED** [review]
@joanna-pagepro Great work, nice one!! I like the Activity feed link appearing when items appear in the activity feed.

1. The user avatar list is not evenly spaced out.
**This PR:**
image
**The design:**
image

3. There is a redundant tooltip when hovering over the description. This could probably be removed.
image

4. You are not able to create a team because you are not able to select a team color. I'm not sure if this was introduced in this PR as I am not able to test `master` at the moment, but in production this is not an issue.
colour-picker

**Comment by joanna-pagepro** [comment]
@arrenv 

1. spacings updated, please check if it's ok now
2. tooltip removed, just as discussed during today's daily call
3. you should be able to create this action now

**Review by arrenv — APPROVED** [review]
Thanks @joanna-pagepro, everything I have checked looks right to me!

File: src/components/v5/common/TeamCardList/TeamCardList.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React, { FC } from 'react';
```
Comment (ArmandoGraterol) [inline src/components/v5/common/TeamCardList/TeamCardList.tsx:1]: Lint errors here

File: src/components/v5/common/TeamCardList/TeamCardList.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import clsx from 'clsx';
+import { motion } from 'framer-motion';
+
+import TeamCard from '../TeamCard/TeamCard';
```
Comment (ArmandoGraterol) [inline src/components/v5/common/TeamCardList/TeamCardList.tsx:5]: This extra `/TeamCard` can be removed

File: src/components/v5/shared/UserAvatars/UserAvatars.tsx [hunk 3]
```diff
@@ -39,19 +41,35 @@ const UserAvatars: FC<UserAvatarsProps> = ({
...
+              />
+              <div
+                className={clsx(
+                  'placeholder absolute inset-0 rounded-full z-[1] bg-gray-50 flex items-center justify-center text-5 text-gray-700 border border-base-white text-center',
+                  {
+                    'border-2': size === 'xms',
+                  },
+                )}
+              >
+                {`+ ${remainingAvatarsCountProp || remainingAvatarsCount}`}
```
Comment (ArmandoGraterol) [inline src/components/v5/shared/UserAvatars/UserAvatars.tsx:68]: What's the difference between the prop and the const in this component? How come we need 2 vars for the count?

File: src/hooks/useActivityData.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import { subDays, startOfDay } from 'date-fns';
+
+import {
+  useGetTotalColonyActionsQuery,
+  useGetTotalColonyDomainActionsQuery,
+} from '~gql';
+import { notNull } from '~utils/arrays';
+
+import useColonyContext from './useColonyContext';
+import { createBaseActionFilter } from './useActivityFeed/helpers';
```
Comment (ArmandoGraterol) [inline src/hooks/useActivityData.ts:10]: Lint error here

**Review by ArmandoGraterol — CHANGES_REQUESTED** [review]
Looking good! I left some small comments and some questions

**Review by melyndav — CHANGES_REQUESTED** [review]
Hi @joanna-pagepro 

Thanks for your work on this.

1. The top right button should be secondary and not primary. It should match all the other buttons across the site that are on the inside pages in the same location. 

2. Some teams are showing with no team members, but when I select 'view team members' from the kebab menu, a team member shows. 

Example on the team page showing no user:
image

But when I select to view members from the kebab this is the result:
image


3. The icons in the kebab menu are showing as 4x14px and should be 16x16 

image

4. Dropdown on mobile isn't the same as Figma - 
image

File: amplify/backend/backend-config.json
[the remark this answers is not in the document]
Comment (joanna-pagepro) [reply amplify/backend/backend-config.json:1]: It's generated automatically, after `npm ci`. I'm not changing anything by myself in this file.

File: src/components/v5/shared/UserAvatars/UserAvatars.tsx (hunk 3 above)
Comment (joanna-pagepro) [reply src/components/v5/shared/UserAvatars/UserAvatars.tsx:68]: The first one - `remainingAvatarsCountProp` - is an optional prop, which you can pass to overwrite the value that is calculated based on avatars to show and items length - `remainingAvatarsCount`
Previous implementation required you to pass the whole members array. This additional prop lets you pass a smaller array with the number of members that should be displayed if there are more members.

**Comment by joanna-pagepro** [comment]
@ArmandoGraterol @melyndav can you check this PR again?

@melyndav I wasn't able to reproduce a bug with missing members. If you will see it again, please add more details.

**Review by ArmandoGraterol — DISMISSED** [review]
Codewise this looks good to me 👍

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @joanna-pagepro 

Thanks for the requested updates! They are all looking to be resolved. 

1. When I open the bottom right hand team, the dropdown menu shows as follows. Is there anyway to improve this? The other dropdowns do not show like this. 

image

2. The placement of the create new team button on mobile should be left aligned. 

image

3. Can we make the width of the dashboard dropdown menu match closer to the Figma design? 
image

4. Can you remove the additional spacing under the description? We will increase the max characters in the team description to 120 from 90.

image

Main component link - https://www.figma.com/file/l1dOM5qiQYwF0ElvKDqqjg/Design-System---Colony-v3?type=design&node-id=1782-43498&mode=design&t=Tnj44be4AzwcSioT-4


5. The extra member circle styling doesn't match Figma- 
image
 
 Figmna -

**Comment by joanna-pagepro** [comment]
@melyndav can you check if it's ok now?

**Comment by joanna-pagepro** [comment]
@melyndav link to team funds page updated. 
@arrenv I can't see any changes in the members array after creating a new team. Can you check if I'm correctly getting members' info? Right now I'm getting members from `useMemberContext` and then I'm checking if roles or reputation items arrays include a domain with a specific team ID. Is there any additional logic to be added here?

**Comment by arrenv** [comment]
> @melyndav link to team funds page updated. @arrenv I can't see any changes in the members array after creating a new team. Can you check if I'm correctly getting members' info? Right now I'm getting members from `useMemberContext` and then I'm checking if roles or reputation items arrays include a domain with a specific team ID. Is there any additional logic to be added here?

The problem here looks to be due to inherited permissions from parent domains not being considered in the members query, this will need to be a separate issue. Thank you for highlighting it.

**Review by arrenv — CHANGES_REQUESTED** [review]
@joanna-pagepro Actually, as @melyndav mentioned, the members are appearing in the members list. You can see them when clicking 'View team members' from the dropdown. Can we use that same query?

image

Also, can this link navigate to the 'Contributor' tab instead, as that is a better representation of this filter.

**Comment by joanna-pagepro** [comment]
@melyndav @arrenv I've pushed some changes. Can you check if it works ok now?

**Review by arrenv — DISMISSED** [review]
@joanna-pagepro Functionally, everything is working as expected. Thank you for the updates.

I am approving, and Mel will do a quick design review, however, can we please hide `View team decisions` button in the drop down for now.

image

**Review by arrenv — APPROVED** [review]

**Review by ArmandoGraterol — DISMISSED** [review]

**Review by melyndav — APPROVED** [review]
Thanks for all your work on this @joanna-pagepro! Approved 👍

**Review by arrenv — APPROVED** [review]
Great work, thank you @joanna-pagepro !

**Pull request merged**